catroid/src/test/java/org/catrobat/catroid/test/visualplacement/VisualPlacementPositionCorrectionTest.java

```diff
@@ -0,0 +1,149 @@
+import org.catrobat.catroid.visualplacement.TimeProvider;
+import org.catrobat.catroid.visualplacement.VisualPlacementTouchListener;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.mockito.ArgumentMatchers.anyFloat;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class VisualPlacementPositionCorrectionTest {
+
+	@Mock
+	ImageView imageView;
+
+	@Mock
+	MotionEvent motionEvent;
+
+	@Mock
+	CoordinateInterface coordinateInterface;
+
+	@Mock
+	TimeProvider timeProvider;
+
+	@InjectMocks
+	VisualPlacementTouchListener listener;
+
+	private final Float initalPositionX = 1.f;
+	private final Float initalPositionY = 1.f;
+	private final Float firstMovementPositionX = 5.f;
+	private final Float firstMovementPositionY = 5.f;
+	private final Float secondMovementPositionX = 6.f;
+	private final Float secondMovementPositionY = 6.f;
+	private final long firstMovementTimestamp = 150;
+	private final long secondMovementTimestamp = 170;
+	private final long actionUpTimestamp = 175;
+
+	@Before
+	public void setUp() {
+		when(timeProvider.elapsedRealtime()).thenReturn((long) 0);
+	}
+
+	public void triggerTouchDownEvent(float startPositionX, float startPositionY) {
+		when(motionEvent.getAction()).thenReturn(MotionEvent.ACTION_DOWN);
+		when(imageView.getX()).thenReturn(startPositionX);
+		when(imageView.getY()).thenReturn(startPositionY);
+		listener.onTouch(imageView, motionEvent, coordinateInterface);
+	}
+
+	public void triggerMovementEvent(long timestamp, float positionX, float positionY) {
+		when(motionEvent.getAction()).thenReturn(MotionEvent.ACTION_MOVE);
+		when(motionEvent.getRawX()).thenReturn(positionX);
+		when(motionEvent.getRawY()).thenReturn(positionY);
+
+		when(timeProvider.elapsedRealtime()).thenReturn(timestamp);
+		when(imageView.getX()).thenReturn(positionX);
+		when(imageView.getY()).thenReturn(positionY);
+		listener.onTouch(imageView, motionEvent, coordinateInterface);
+	}
+
+	public void triggerTouchUpEvent(long timestamp) {
+		when(timeProvider.elapsedRealtime()).thenReturn(timestamp);
+		when(motionEvent.getAction()).thenReturn(MotionEvent.ACTION_UP);
+		listener.onTouch(imageView, motionEvent, coordinateInterface);
+	}
+
+	@Test
+	public void testTouchCorrection() {
+		triggerTouchDownEvent(initalPositionX, initalPositionY);
+		triggerMovementEvent(firstMovementTimestamp, firstMovementPositionX, firstMovementPositionY);
+		triggerMovementEvent(secondMovementTimestamp, secondMovementPositionX, secondMovementPositionY);
+		triggerTouchUpEvent(actionUpTimestamp);
+
+		verify(imageView).setX(firstMovementPositionX);
+		verify(imageView).setY(firstMovementPositionY);
+		verify(imageView, times(3)).setX(anyFloat());
```

Comment:
I don't think that this is doing what you want, though I am not sure. From my understanding, this just checks that getX was invoked with these values at some time, but not that they were necessarily the final invocations. Please change that so that it is made sure that the the final values are the ones from the first movement. This may then be clearer for others who want to understand the test. 

The number of invocations test in the next line  is OK too for this specific implementation and test case.